code/modules/fluids/fluid_pipe_machinery.dm

```diff
@@ -0,0 +1,521 @@
+			src.reagents.trans_to(P, src.amount)
+			src.visible_message("[src] ejects a pill.")
+		if("vials")
+			var/obj/item/reagent_containers/glass/vial/plastic/V = new(get_turf(src))
+			src.reagents.trans_to(V, src.amount)
+			src.visible_message("[src] ejects a vial.")
+		if("patches")
+			var/obj/item/reagent_containers/patch/P = new(get_turf(src))
+			src.reagents.trans_to(P, src.amount)
+			src.visible_message("[src] ejects a patch.")
+
+/obj/machinery/fluid_machinery/unary/dispenser/proc/set_amount(var/datum/mechanicsMessage/input)
+	var/newamount = text2num_safe(input.signal)
+	if (!newamount)
+		return
+	src.amount = round(clamp(newamount, src.min, src.max), QUANTIZATION_UNITS)
+
+/obj/machinery/fluid_machinery/unary/dispenser/proc/set_amount_manual(obj/item/W, mob/user)
+	var/inp = tgui_input_number(user, "Please enter dispense amount (Will round to [QUANTIZATION_UNITS]):", "Dispense Amount", src.amount, src.max, src.min)
+	if (!inp) return
+	src.amount = round(inp, QUANTIZATION_UNITS)
+
+/obj/machinery/fluid_machinery/unary/dispenser/proc/set_type(obj/item/W, mob/user)
+	var/inp = tgui_input_list(user, "Select a type to output.", "Dispense Type", src.itemlist)
+	src.itemtodispense = (inp in src.itemlist) ? inp : src.itemtodispense
+
+/obj/machinery/fluid_machinery/unary/dispenser/proc/set_automatic(obj/item/W, mob/user)
+	src.automatic = !src.automatic
+	boutput(user, SPAN_NOTICE("Automatic mode is now set to [src.automatic ? "true" : "false"]."))
+
+/obj/machinery/fluid_machinery/unary/dispenser/New()
+	..()
+	src.create_reagents(src.max)
+	AddComponent(/datum/component/mechanics_holder)
+	SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_INPUT,"dispense now", PROC_REF(dispense))
+	SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_CONFIG,"Set Dispense Amount", PROC_REF(set_amount_manual))
+	SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_CONFIG,"Set Dispense Type", PROC_REF(set_type))
+	SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_CONFIG,"Set Automatic dispensing", PROC_REF(set_automatic))
+
+/obj/machinery/fluid_machinery/unary/dispenser/get_desc()
+		. += "<br>[SPAN_NOTICE("Automatic: [src.automatic ? "true" : "false"]. Dispense Amount: [src.amount]. Dispensing: [src.itemtodispense]")]"
+
+/obj/machinery/fluid_machinery/unary/dispenser/process()
+	if (!src.network) return
+	var/datum/reagents/fluid = src.pull_from_network(src.network, src.max)
+	fluid.trans_to(src, src.max)
+	src.push_to_network(src.network, fluid)
+	src.reagents.handle_reactions()
+	SEND_SIGNAL(src,COMSIG_MECHCOMP_TRANSMIT_SIGNAL, "[src.reagents.total_volume]")
+	if (!src.automatic)
+		return
+	src.dispense()
+
+/obj/machinery/fluid_machinery/unary/node
+	name = "Node"
+	desc = "Used for connecting non-fluid machinery to fluid pipes, AKA, YOU SHOULDNT SEE THIS."
+	invisibility = INVIS_ALWAYS
+
+/obj/machinery/fluid_machinery/unary/node/ex_act()
+	return
+
+/obj/machinery/fluid_machinery/unary/node/meteorhit()
+	return
+
+/obj/machinery/fluid_machinery/unary/node/updateHealth()
+	return
```

Comment:
?

Reply:
invisible object used for connecting non fluid stuff to fluid stuff.